packages/ra-ui-materialui/src/list/datagrid/useDatagridContext.ts

```diff
@@ -0,0 +1,28 @@
+import { useContext, useMemo } from 'react';
+import { DatagridProps } from './Datagrid';
+import DatagridContext, { DatagridContextValue } from './DatagridContext';
+import merge from 'lodash/merge';
+import { warning } from 'ra-core';
+
+export const useDatagridContext = (
+    props?: DatagridProps
+): DatagridContextValue => {
+    const context = useContext(DatagridContext);
+
+    warning(
```

Comment:
As the default context can be enough, and as users can override the context with props, we don't want to log a warning here.

Reply:
OK